base/error.jl

```diff
@@ -218,6 +219,9 @@ julia> @assert isodd(3) "What even are numbers?"
 ```
 """
 macro assert(ex, msgs...)
+    enable_asserts = !isdefined(Main, :Base) || Main.Base.julia_debug
```

Comment:
Why do we need the isdefined check here? The macro itself should be able to see this unconditionally